src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -1687,7 +1687,7 @@ def __init__(self, config):
                 "instantiate the model as follows: `Wav2Vec2ForCTC.from_pretrained(..., vocab_size=vocab_size)`. "
                 "or define `vocab_size` of your model's configuration."
             )
-        self.lm_head = nn.Linear(config.hidden_size, config.vocab_size)
+        self.lm_head = nn.Linear(config.output_hidden_size if config.add_adapter else config.hidden_size, config.vocab_size)
```

Comment:
```suggestion
        output_hidden_size = config.output_hidden_size if config.add_adapter else config.hidden_size
        self.lm_head = nn.Linear(output_hidden_size, config.vocab_size)
```

It's a bit more readable